Zend/zend_ast.h

```diff
@@ -313,7 +313,8 @@ typedef void (*zend_ast_apply_func)(zend_ast **ast_ptr, void *context);
 ZEND_API void zend_ast_apply(zend_ast *ast, zend_ast_apply_func fn, void *context);
 
 static zend_always_inline size_t zend_ast_size(uint32_t children) {
-	return sizeof(zend_ast) - sizeof(zend_ast *) + sizeof(zend_ast *) * children;
+	ZEND_ASSERT(children > 0);
```

Comment:
`0` is indeed allowed.

Reply:
Build fails on debug mode so yeah.. if children is 0 then at least on this compiler it returns the same value.so I 'll remove the new assertion